test/tests/functional/pbs_cpuset_reliable_job_startup.py

```diff
@@ -0,0 +1,162 @@
+        The released numa node can be used in another job.
+        """
+
+        # instantiate execjob_launch hook
+        hook_event = "execjob_launch"
+        hook_name = "launch"
+        a = {'event': hook_event, 'enabled': 'true'}
+        stime = int(time.time())
+        self.server.create_import_hook(hook_name, a, self.launch_hook_body)
+
+        # Check mom logs that the launch hook got propagated
+        msg = "Hook;launch.PY;copy hook-related file request received"
+        self.mom.log_match(msg, starttime=stime, interval=16, n=10)
+
+        # Submit job1
+        j = Job(TEST_USER, {
+            ATTR_l + '.select': '2:ncpus=%s:mem=2gb' % self.vnA_ncpus,
+            ATTR_l + '.place': 'free',
+            ATTR_W: 'tolerate_node_failures=job_start'})
+        stime = int(time.time())
+        jid = self.server.submit(j)
+
+        # Check the exec_vnode while in substate 41
+        self.server.expect(JOB, {ATTR_substate: '41'}, id=jid)
+        self.server.expect(JOB, 'exec_vnode', id=jid, op=SET)
+        job_stat = self.server.status(JOB, id=jid)
+        execvnode1 = job_stat[0]['exec_vnode']
+        self.logger.info("initial exec_vnode: %s" % execvnode1)
+        initial_vnodes = execvnode1.split('+')
+
+        # Check the exec_vnode after job is in R state, substate 42
+        self.server.expect(JOB, {ATTR_substate: '42'}, interval=7, id=jid)
+        self.server.expect(JOB, 'exec_vnode', id=jid, op=SET)
+        job_stat = self.server.status(JOB, id=jid)
+        execvnode2 = job_stat[0]['exec_vnode']
+        self.logger.info("pruned exec_vnode: %s" % execvnode2)
+
+        # Check for msg in mom logs
+        msg = "new_cpuset:  setting altid to CPU set named /PBSPro/%s" % jid
+        self.mom.log_match(msg, starttime=stime, regexp=True, n=10)
+
+        # Check mom logs for pruned from and pruned to messages
+        self.mom.log_match("Job;%s;pruned from exec_vnode=%s" % (
+            jid, execvnode1), starttime=stime, n=10)
+        self.mom.log_match("Job;%s;pruned to exec_vnode=%s" % (
+            jid, execvnode2), starttime=stime, n=10)
+
+        # Find out the released vnode
+        if initial_vnodes[0] == execvnode2:
+            execvnodeB = initial_vnodes[1]
+        else:
+            execvnodeB = initial_vnodes[0]
+        vnodeB = execvnodeB.split(':')[0].split('(')[1]
+        self.logger.info("released vnode: %s" % vnodeB)
+
+        # Submit job2 requesting the released vnode, job runs
+        j2 = Job(TEST_USER, {
+            ATTR_l + '.select': '1:ncpus=%s:mem=2gb:vnode=%s' % (
+                self.vnA_ncpus, vnodeB)})
+        stime = int(time.time())
+        jid2 = self.server.submit(j2)
+        self.server.expect(JOB, {ATTR_state: 'R'}, interval=7, id=jid2)
+
+        # Check for msg in mom superior logs
+        msg2 = "new_cpuset:  setting altid to CPU set named /PBSPro/%s" % jid2
+        self.mom.log_match(msg2, starttime=stime, regexp=True, n=10)
```

Comment:
Same here, I don't think you need regexp=True.

Reply:
@bayucan removed regexp.
@anamikau this log message tells us that there is a cpuset for the job.